### PR TITLE
Testspec Update: ML-KEM, ML-DSA and SLH-DSA

### PR DESCRIPTION
**Pull request by FAlbertDev**
This PR contains the testspec adaptions for the three new FIPS 20X algorithms. I kept the level of detail roughly the same.

### CODE REVIEW

File: docs/testspec/src/12_pubkey_kem.rst [hunk 1]
```diff
@@ -109,53 +109,59 @@ tests are in *src/tests/data/pubkey/frodokem_kat.vec*.
...
 
    +------------------------+-------------------------------------------------------------------------+
-   | **Test Case No.:**     | PKENC-KYBER-1                                                           |
+   | **Test Case No.:**     | PKENC-ML-KEM-1                                                          |
    +========================+=========================================================================+
    | **Type:**              | Known Answer Tests                                                      |
    +------------------------+-------------------------------------------------------------------------+
-   | **Description:**       | Uses the KAT vectors of Kyber's reference implementation as specified   |
-   |                        | in the NIST submission                                                  |
+   | **Description:**       | Uses the KAT vectors for ML-KEM.                                        |
```
Comment (atreiber94) [inline docs/testspec/src/12_pubkey_kem.rst:142]: This uses the Kris Kwiatkowski test for ML-KEM. For Kyber it is still the reference KATs

File: docs/testspec/src/14_pubkey_sig.rst [hunk 2]
```diff
@@ -395,59 +395,59 @@ described here in the following.
...
 
-Dilithium
----------
+ML-DSA
+------
 
-The implementation is tested for correctness using the Known Answer Test vectors
-demanded by the NIST submission and provided by the reference implementation.
+The implementation is tested for correctness using Known Answer Test vectors
+generated using an implementation associated with the ACVP project.
```
Comment (atreiber94) [inline docs/testspec/src/14_pubkey_sig.rst:402]: the implementation is from https://github.com/mjosaarinen/py-acvp-pqc/tree/main which has been tested against the ACVP project.

For ML-DSA, we do not use any ACVP vectors in Botan (though I just noticed that it is stated as such in the ML-DSA commit message...)

File: docs/testspec/src/14_pubkey_sig.rst [hunk 3]
```diff
@@ -520,6 +519,37 @@ the algorithm parameters, namely *4x4\_Deterministic*, *6x5\_Deterministic*,
    |                        | #. Ensure that these decoded keys work for signing and verifying.       |
    +------------------------+-------------------------------------------------------------------------+
 
+.. table::
+   :class: longtable
+   :widths: 20 80
+
+   +------------------------+-------------------------------------------------------------------------+
+   | **Test Case No.:**     | PKENC-ML-DSA-2                                                          |
```
Comment (atreiber94) [inline docs/testspec/src/14_pubkey_sig.rst:527]:
```suggestion
   | **Test Case No.:**     | PKSIG-ML-DSA-2                                                          |
```

Comment (atreiber94) [reply docs/testspec/src/14_pubkey_sig.rst:527]: Also, this test description is for encaps/decaps but ML-DSA is a DSA

File: docs/testspec/src/14_pubkey_sig.rst [hunk 4]
```diff
@@ -1523,17 +1553,17 @@ constraints for this test case are:
...
-SPHINCS+
---------
+SLH-DSA
+-------
 
-The implementation is tested for correctness using the Known Answer Test vectors
-demanded by the NIST submission and provided by the reference implementation.
-Given SPHINCS+' performance characteristics, each supported algorithm
+The implementation is tested for correctness using Known Answer Test vectors
+generated using an implementation associated with the ACVP project.
```
Comment (atreiber94) [inline docs/testspec/src/14_pubkey_sig.rst:1560]: See ML-DSA comment (same repo, same procedure)

File: docs/testspec/src/12_pubkey_kem.rst [hunk 5]
```diff
@@ -109,53 +109,59 @@ tests are in *src/tests/data/pubkey/frodokem_kat.vec*.
...
 
-Kyber
-~~~~~
+ML-KEM
+~~~~~~
 
-The implementation is tested for correctness using the Known Answer Test vectors
-demanded by the NIST submission and provided by the reference implementation.
+The implementation is tested for correctness using Known Answer Test vectors
+generated using an implementation associated with the ACVP project.
```
[the remark this answers is not in the document]
Comment (FAlbertDev) [reply docs/testspec/src/12_pubkey_kem.rst:116]: Nice, I'll add the reference 👍

File: docs/testspec/src/12_pubkey_kem.rst (hunk 1 above)
Comment (FAlbertDev) [reply docs/testspec/src/12_pubkey_kem.rst]: I think it suffices here to speak about KAT vectors in general for ML-KEM/Kyber. The source of the vectors is stated above. Which files are used is stated below.

**Comment by FAlbertDev** [comment]
Thanks for the review! I applied your suggestions, @atreiber94.

**Comment by reneme** [comment]
@FAlbertDev rebased to `main`, hoping that the `:srcref:` now work.

**Pull request merged**